src/cluster.c

```diff
@@ -108,6 +108,27 @@ ConnectionType *connTypeOfCluster(void) {
     return connectionTypeTcp();
 }
 
+void delUnownedKeys(void) {
+    if (!server.cluster_enabled) {
+        return;
+    }
+    clusterNode *primary;
+    if (clusterNodeIsPrimary(server.cluster->myself)) {
+        primary = server.cluster->myself;
+    } else {
+        primary = server.cluster->myself->replicaof;
+    }
```

Comment:
There is a method to fetch the primary i.e. `clusterNodeGetPrimary`, we could use that here.